diff -ur scorched3d-43.3.orig/src/client/client/SecureID.cpp scorched3d-43.3/src/client/client/SecureID.cpp
--- scorched3d-43.3.orig/src/client/client/SecureID.cpp	2010-07-20 02:28:59.000000000 +0200
+++ scorched3d-43.3/src/client/client/SecureID.cpp	2012-11-16 13:38:27.545520369 +0100
@@ -75,6 +75,7 @@
 
 #include <sys/ioctl.h>
 #include <net/if.h>
+#include <unistd.h>
 
 std::string SecureID::GetPrivateKey(void)
 {
